ground/src/main/java/com/google/android/ground/ui/offlineareas/selector/OfflineAreaSelectorFragment.kt

```diff
@@ -75,4 +75,26 @@ class OfflineAreaSelectorFragment : AbstractMapContainerFragment() {
   }
 
   override fun getMapViewModel(): BaseMapViewModel = viewModel
+
+  private fun showDownloadProgressDialog(isVisible: Boolean) {
+    val dialogComposeView =
+      ComposeView(requireContext()).apply {
+        setContent {
+          val openAlertDialog = remember { mutableStateOf(isVisible) }
+          val progress = viewModel.downloadProgress.observeAsState(0f)
+          when {
+            openAlertDialog.value -> {
+              AppTheme {
+                DownloadProgressDialog(
+                  progress = progress.value,
+                  onDismiss = { openAlertDialog.value = false },
```

Comment:
Can you please add a TODO (and file a bug) to also cancel the download request if the cancel button is clicked?